gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -243,6 +247,10 @@ synchronized void addDag(Dag<JobExecutionPlan> dag) throws IOException {
     submitEventsAndSetStatus(dag);
   }
 
+  synchronized void addDag(Dag<JobExecutionPlan> dag) throws IOException {
```

Comment:
Is this for backwards compatibility? I'm curious why we need an extra function for implicit persistence.

Reply:
Mainly just to provide a default value since some unit tests call this, but yeah maybe it's better to update all the callers.